pfcpiface/pfcpiface.go

```diff
@@ -17,61 +16,56 @@ import (
 )
 
 var (
-	configPath = flag.String("config", "upf.json", "path to upf config")
-	simulate   = simModeDisable
-	pfcpsim    = flag.Bool("pfcpsim", false, "simulate PFCP")
+	simulate = simModeDisable
+	pfcpsim  = flag.Bool("pfcpsim", false, "simulate PFCP")
 )
 
 func init() {
 	flag.Var(&simulate, "simulate", "create|delete|create_continue simulated sessions")
-	// Set up logger
-	log.SetReportCaller(true)
-	log.SetFormatter(&log.TextFormatter{
-		FullTimestamp: true,
-	})
 }
 
-func main() {
-	// cmdline args
-	flag.Parse()
+type PFCPIface struct {
+	conf Conf
 
-	// Read and parse json startup file.
-	conf, err := LoadConfigFile(*configPath)
-	if err != nil {
-		log.Fatalln("Error reading conf file:", err)
-	}
-
-	log.SetLevel(conf.LogLevel)
+	fp  fastPath
+	upf *upf
+}
 
-	log.Infof("%+v", conf)
+func NewPFCPIface(conf Conf) *PFCPIface {
+	pfcpIface := &PFCPIface{
+		conf: conf,
+	}
 
-	var fp fastPath
 	if conf.EnableP4rt {
-		fp = &UP4{}
+		pfcpIface.fp = &UP4{}
 	} else {
-		fp = &bess{}
+		pfcpIface.fp = &bess{}
 	}
 
-	upf := NewUPF(&conf, fp)
+	pfcpIface.upf = NewUPF(&conf, pfcpIface.fp)
+
+	return pfcpIface
+}
 
+func (p *PFCPIface) Run() {
```

Comment:
Could this return an `error`? Doesn't mean we have to fix all code in here yet, just return the `err`s we have now.

Reply:
the functions that we call from `Run()` don't return any errors now. I could leave `err` empty, but then the linter doesn't let it go.

Reply:
ok.
This really needs to be on the fastpath interface level first.